### PR TITLE
cclient: added logger.

### LINKED ISSUE #222: Add logger in cclient.
Use `utils/logger_helper` as logger facility.

### PR DESCRIPTION
**Pull request by oopsmonk**
This PR fixed #222 

# Test Plan:
CI must pass

### CODE REVIEW

File: cclient/iota_api.c [hunk 1]
```diff
@@ -28,6 +36,8 @@ iota_api_result_t iota_api_get_node_info(
 
   result = iota_service_query(service, req_buff, res_buff);
   if (result.error != RC_OK) {
+    log_error(CCLIENT_CORE_LOGGER_ID, "[%s:%d] HTTP request filed. %x\n",
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 2]
```diff
@@ -63,6 +75,8 @@ iota_api_result_t iota_api_get_neighbors(
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 3]
```diff
@@ -99,6 +116,8 @@ iota_api_result_t iota_api_add_neighbors(
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 4]
```diff
@@ -134,6 +156,8 @@ iota_api_result_t iota_api_remove_neighbors(
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 5]
```diff
@@ -169,6 +196,8 @@ iota_api_result_t iota_api_get_tips(const iota_http_service_t* const service,
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 6]
```diff
@@ -204,6 +236,8 @@ iota_api_result_t iota_api_find_transactions(
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 7]
```diff
@@ -239,6 +276,8 @@ iota_api_result_t iota_api_get_trytes(const iota_http_service_t* const service,
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 8]
```diff
@@ -274,6 +316,8 @@ iota_api_result_t iota_api_get_inclusion_states(
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 9]
```diff
@@ -311,6 +358,8 @@ iota_api_result_t iota_api_get_balances(
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 10]
```diff
@@ -349,6 +401,8 @@ iota_api_result_t iota_api_get_transactions_to_approve(
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 11]
```diff
@@ -386,6 +443,8 @@ iota_api_result_t iota_api_attach_to_tangle(
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 12]
```diff
@@ -429,6 +492,8 @@ iota_api_result_t iota_api_broadcast_transactions(
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 13]
```diff
@@ -461,6 +529,8 @@ iota_api_result_t iota_api_store_transactions(
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c [hunk 14]
```diff
@@ -495,6 +568,8 @@ iota_api_result_t iota_api_check_consistency(
...
```
Comment (tsvisabo) [inline cclient/iota_api.c]: failed

File: cclient/iota_api.c (hunk 1 above)
Comment (tsvisabo) [reply cclient/iota_api.c]: btw, can you create a `HTTP_REQUEST_FAILED = "[%s:%d] HTTP request filed. %x\n"` ?

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: common/errors.h [hunk 15]
```diff
@@ -370,6 +370,9 @@ enum retcode_t {
 
 typedef enum retcode_t retcode_t;
 
+#define STR_OK "OK"
+
+// Cclient Module
```
Comment (tsvisabo) [inline common/errors.h]: can we have those in a different file? :)

**Review by tsvisabo — CHANGES_REQUESTED** [review]

File: common/errors.h (hunk 15 above)
Comment (oopsmonk) [reply common/errors.h]: `error_strings.h`?

Comment (tsvisabo) [reply common/errors.h]: that's what i thought :)

**Comment by th0br0** [comment]
If we put them in a separate file, we need to make sure that namespaces & ideally value spaces don't overlap. 
Could you write up a markdown file describing the namespacing for errors as part of the PR so we can have something to start with for future modules?

**Comment by oopsmonk** [comment]
the compiler will tell you if it has overlap. before this modification, we don't know where it is.

**Review by tsvisabo — APPROVED** [review]

**Pull request merged**